core/src/main/java/com/taobao/arthas/core/view/ObjectView.java

```diff
@@ -578,7 +578,13 @@ else if (Date.class.isInstance(obj)) {
                     appendStringBuilder(buf, format("@%s[%s]", className, obj));
                 } else {
                     appendStringBuilder(buf, format("@%s[", className));
-                    final Field[] fields = obj.getClass().getDeclaredFields();
+                    List<Field> fields = new ArrayList<Field>();
```

Comment:
This can be very dangerous depend on the object you are watching.
The class hierarchy may be very deep, for example org.apache.dubbo.config.spring.ServiceBean, has very deep inheritance, and for each super class, there are many fields.
Watching such an object with parent fields, may be very dangerous because it may quickly consume lots of memory.

I'd to see more defensive coding here:

1.  disable this by default, user should add an option in order to enable it.
2.  add a limit on how many fields can we get at most. This should be divided into two parts, 
    1.  limit the depth of the iteration, which means the while loop should stop if reaches the limit. 
    2.  limit the max number of fields we can get

Generally, I prefer to recommend user to use `ognl experssion` to access the parent's fields the want.

Reply:
You can just use the ognl expression `target.getXXX()` to get parent's fields. `target` means `this`, and XXX is the name of the parent's fields